murageden/Retrofit-GithubApi
# #158787482 feature(dependency versions): manage build cache and dependency versions

## What does this PR do?
- Improve build speeds with gradle build cache
- Manage dependency versions in build.gradle file

## Description of Task to be completed?
- Improve build speeds with gradle build cache
- Manage dependency versions in build.gradle file

## How should this be manually tested?

- Clone the repo.
- Set up the project with android studio.
- Go to gradle.properties and remove `org.gradle.caching=true` then sync gradle and see the time it takes to run the build.
- Return `org.gradle.caching=true` and run the build twice to see the difference.

## What are the relevant pivotal tracker stories?
[158787096](https://www.pivotaltracker.com/story/show/158787096)
[158787482](https://www.pivotaltracker.com/story/show/158787482) 

## Screenshots (if appropriate)?
- Before caching
![image](https://user-images.githubusercontent.com/22506247/44198317-bbc90c80-a149-11e8-953d-9f6b92df1da6.png)
- After caching
![image](https://user-images.githubusercontent.com/22506247/44198286-a8b63c80-a149-11e8-9ac2-b37f69c714c7.png)


## Questions:
N/A
